### PR TITLE
Use golang 1.19

### PR DESCRIPTION
**Pull request by pavolloffay**
Signed-off-by: Pavol Loffay <p.loffay@gmail.com>

### CODE REVIEW

File: .github/workflows/continuous-integration.yaml [hunk 1]
```diff
@@ -35,15 +35,15 @@ jobs:
...
+        go-version: 1.19
     - name: Check out code into the Go module directory
       uses: actions/checkout@v3
 
     - name: Lint
       uses: golangci/golangci-lint-action@v3
       with:
         args: -v
-        version: v1.45
+        version: v1.47
```
Comment (pavolloffay) [inline .github/workflows/continuous-integration.yaml:46]: Go 1.19 support for golangci https://github.com/golangci/golangci-lint/pull/3037

Comment (pavolloffay) [reply .github/workflows/continuous-integration.yaml]: I cannot get the linter to work. see as well https://github.com/golangci/golangci-lint/issues/2374#issuecomment-1203115918

**Comment by pavolloffay** [comment]
I am happy to merge this and book an issue to update golangci once it supports golang 1.19

**Review by frzifus — COMMENTED** [review]
mh.. now iam confused by the diff? The ci worked, but there is no downgrade in the files?

**Comment by frzifus** [comment]
> >  but there is no downgrade in the files?
> 
> what downgrade? 

This one?  https://github.com/open-telemetry/opentelemetry-operator/commit/9e6d86b7a754f17d49e3bfb2c36887baa60f563e

**Comment by pavolloffay** [comment]
There is no downgrade, I have reverted it back to the original version.

File: pkg/collector/adapters/config_validate.go [hunk 2]
```diff
@@ -18,9 +18,9 @@ import (
 	"github.com/go-logr/logr"
 )
 
-//Following Otel Doc: Configuring a receiver does not enable it. The receivers are enabled via pipelines within the service section.
-//GetEnabledReceivers returns all enabled receivers as a true flag set. If it can't find any receiver, it will return a nil interface.
-func GetEnabledReceivers(logger logr.Logger, config map[interface{}]interface{}) map[interface{}]bool {
+// Following Otel Doc: Configuring a receiver does not enable it. The receivers are enabled via pipelines within the service section.
```
Comment (pavolloffay) [inline pkg/collector/adapters/config_validate.go:21]: lint was complaining that there was no space

File: pkg/collector/reconcile/horizontalpodautoscaler_test.go [hunk 3]
```diff
@@ -17,7 +17,7 @@ package reconcile
 import (
 	"context"
 	"fmt"
-	"io/ioutil"
```
Comment (pavolloffay) [inline pkg/collector/reconcile/horizontalpodautoscaler_test.go:20]: `io/ioutil` is deprecated

**Comment by pavolloffay** [comment]
PR is ready to be merged

**Comment by pavolloffay** [comment]
cc) @VineethReddy02 @jpkrohling @yuriolisa

**Comment by yuriolisa** [comment]
@pavolloffay, could you please rebase it?

**Review by VineethReddy02 — APPROVED** [review]

**Comment by pavolloffay** [comment]
PR rebased.

**Pull request merged**